src/controllers/products/addProduct.ts

```diff
@@ -0,0 +1,24 @@
+import { Response, Request } from 'express';
+import { Product } from '../../database/productSchema';
+
+const addProduct = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const { title, image, rating, price, category } = req.body;
+    const isExistedProduct = await Product.findOne({ title });
+    if (isExistedProduct) {
+      res.status(400).json({ msg: 'Product Is Existing' });
```

Comment:
use custom error instead of sending json

Reply:
done